src/frontend/sharedComponents/HookFormTextInput.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import {
+  Control,
+  Controller,
+  FieldValues,
+  Path,
+  useWatch,
+  ValidationRule,
+  useFormState,
+  UseControllerProps,
+} from 'react-hook-form';
+import {TextInput as RNTextInput, StyleSheet, View} from 'react-native';
+import {BLACK, LIGHT_GREY, RED} from '../lib/styles';
+import {ErrorIcon} from './icons';
+import {ViewStyleProp} from '../sharedTypes';
+import {Text} from './Text';
+
+type TextInputProps<InputFields extends FieldValues> = {
+  containerStyle?: ViewStyleProp;
+  showCharacterCount?: boolean;
+  control: Control<InputFields>;
+} & Omit<
+  React.ComponentProps<typeof RNTextInput>,
+  'value' | 'onBlur' | 'onChange'
```

Comment:
I know we talked about only using the placeholder prop, but we actually use other props in other components, so I think it would be good to have access to them. Perhaps we should also omit the styling prop because it could be a little confusing since there is technically a `view` around it that controls certain aspects of the style.

Reply:
if that's the case, i would suggest using `Pick` e.g. `Pick<React.ComponentProps<typeof RNTextInput>, 'placeholder'>`

Reply:
sorry, i wasnt the clearest with my comment. Im actually saying that "placeholder" is not the only one we use. For example in the manual GPS form, we use `keyboardType`, `accessibilityLabel`, and `underlineColorAndroid`. So I think we should expose them to the user. Im using omit because `  'value' | 'onBlur' | 'onChange' ` are handled by the Controller, so those are the ones we do not want to expose

Reply:
ah yeah if you think there will be a variety that are needed, then keeping the Omit makes sense. You can use Pick to get multiple values, but if there's an indefinite list then it's just too verbose 😄